tests/cpp/test_alias_analysis.cpp

```diff
@@ -293,4 +294,240 @@ TEST_F(AliasAnalysisTest, AliasForReshardingExprs) {
   EXPECT_TRUE(analysis.getRoot(out) == in);
 }
 
+TEST_F(AliasAnalysisTest, View_NotInvolvingDid) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  constexpr int kNumDevices = 4;
+  auto mesh = DeviceMesh::createForNumDevices(kNumDevices);
+
+  TensorView* in = makeContigConcreteTensor({-1, -1, 768});
+  TensorView* out = reshape(in, {mul(size(in, 0), size(in, 1)), size(in, 2)});
+
+  for (auto* tv : {in, out}) {
+    tv->setDeviceMesh(mesh);
+    tv->outer_split(-1, kNumDevices);
+    tv->axis(-2)->parallelize(ParallelType::DIDx);
+    tv->setAllocationDomain(tv->getLoopDomain(), true);
+  }
+
+  fusion.addInput(in);
+  fusion.addOutput(out);
+
+  at::Tensor in_tensor =
+      at::randn({2, 3, 768 / kNumDevices}, at::Device(at::kCUDA));
+  KernelArgumentHolder args({in_tensor});
+
+  // Concretization is needed for reshapes involving dynamic sizes.
+  DynamicTransform::concretizeFusion(&fusion, args);
+  ASSERT_EQ(fusion.inputs().size(), 1);
+  ASSERT_EQ(fusion.outputs().size(), 1);
+  // Concretization updates fusion's inputs and outputs.
+  in = fusion.inputs()[0]->as<TensorView>();
+  out = fusion.outputs()[0]->as<TensorView>();
+
+  AliasAnalysisResult analysis = findAliases(&fusion);
+  EXPECT_TRUE(analysis.getRoot(out) == in);
+}
+
+TEST_F(AliasAnalysisTest, View_DidInFront) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  constexpr int kNumDevices = 4;
+  auto mesh = DeviceMesh::createForNumDevices(kNumDevices);
+
+  TensorView* in = makeContigConcreteTensor({-1, -1, 768});
+  TensorView* out = reshape(in, {mul(size(in, 0), size(in, 1)), size(in, 2)});
+
+  for (auto* tv : {in, out}) {
+    tv->setDeviceMesh(mesh);
+    tv->outer_split(-1, kNumDevices);
+    tv->axis(-2)->parallelize(ParallelType::DIDx);
+    tv->reorder({{-2, 0}});
+    tv->setAllocationDomain(tv->getLoopDomain(), true);
+  }
+
+  fusion.addInput(in);
+  fusion.addOutput(out);
+
+  at::Tensor in_tensor =
+      at::randn({2, 3, 768 / kNumDevices}, at::Device(at::kCUDA));
+  KernelArgumentHolder args({in_tensor});
+
+  // Concretization is needed for reshapes involving dynamic sizes.
+  DynamicTransform::concretizeFusion(&fusion, args);
+  ASSERT_EQ(fusion.inputs().size(), 1);
+  ASSERT_EQ(fusion.outputs().size(), 1);
+  // Concretization updates fusion's inputs and outputs.
+  in = fusion.inputs()[0]->as<TensorView>();
+  out = fusion.outputs()[0]->as<TensorView>();
+
+  AliasAnalysisResult analysis = findAliases(&fusion);
+  EXPECT_TRUE(analysis.getRoot(out) == in);
+}
+
+TEST_F(AliasAnalysisTest, View_DidOnMerge) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  constexpr int kNumDevices = 4;
+  auto mesh = DeviceMesh::createForNumDevices(kNumDevices);
+
+  TensorView* in = makeContigConcreteTensor({-1, -1, 12, 64});
+  TensorView* out =
+      reshape(in, {size(in, 0), size(in, 1), IrBuilder::create<Val>(768)});
```

Comment:
QQ: is constant size a requirement for DID?

Reply:
No but... For a reshape that splits/merges a sharded dimension, constant sizes are useful for isResharding to reason about divisibility. For example, consider a reshape from `logical=[h], loop=[d, h/d]` to `logical=[a, h/a], loop=[d, a/d, h/a]`. If `h` is symbolic, the two `d`s can map to different indices and the reshape may be resharding.